### PR TITLE
feat(Komga): Add Komga

### PR DESCRIPTION
**Pull request by ComputerByte**
!! Depends on #657 

## Description
Add Komga to box installs

## Fixes issues: 
- Fixes Nothing

## Proposed Changes:
- Add Komga to the stack

## Change Categories

- New feature / Application

## Checklist
- [x] Docs have been made OR are not necessary
    - PR link: liaralabs/docs.swizzin.ltd#72
- [x] Changes to panel have been made OR are not necessary
    - PR link: liaralabs/swizzin_dashboard#36
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Code conforms to project structure [(See more)](https://swizzin.ltd/dev/structure)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have functionally tested the changes being introduced

## Test scenarios
- Tested on AMD64

### Architectures
|   			| `amd64` 	| `armhf` 	| `arm64` 	| Unspecified 	|
|--------		|-------- 	|-------- 	|-------- 	|----------		|
| Focal 		|			|			|			|				|
| Bionic		|			|			|			|				|
| Buster		|	✅		|			|			|				|
| Stretch		|			|			|			|				|
| Raspbian  	|			|			|			|				|

### 🛠🛠 TODO
- [x] Updater script

### ❌❌ Currently failing

### CODE REVIEW

File: scripts/install/komga.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+. /etc/swizzin/sources/globals.sh
+. /etc/swizzin/sources/functions/utils
+
+# Script by @ComputerByte 
+# For Komga Installs
+
+# Log to Swizzin.log
+export log=/root/logs/swizzin.log
+touch $log
```
Comment (flying-sausages) [inline scripts/install/komga.sh:10]: This is actually handled with `globals.sh`

File: scripts/install/komga.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  proxy_pass        http://127.0.0.1:8080/komga;
+}
+NGX
+# Reload nginx
+systemctl reload nginx
+echo_progress_done "Nginx config applied"
+fi
+
+
+touch /install/.komga.lock   >> $log 2>&1
```
Comment (flying-sausages) [inline scripts/install/komga.sh:55]:
```suggestion
touch /install/.komga.lock
```

File: scripts/install/komga.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+systemctl restart panel   >> $log 2>&1
```
Comment (flying-sausages) [inline scripts/install/komga.sh:56]: I think this is leftover from when you were `>>`ing into the panel profiles

File: scripts/install/komga.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+. /etc/swizzin/sources/globals.sh
```
Comment (flying-sausages) [inline scripts/install/komga.sh:2]: This is sourced in `box` for you

File: scripts/install/komga.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+. /etc/swizzin/sources/functions/utils
```
Comment (flying-sausages) [inline scripts/install/komga.sh:3]: Not sure if this is here for the user functions, but those are now in globals too

File: scripts/install/komga.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    cat > /etc/nginx/apps/komga.conf <<- NGX
+location /komga {
+  include /etc/nginx/snippets/proxy.conf;
+  proxy_pass        http://127.0.0.1:8080/komga;
+}
+NGX
+# Reload nginx
+systemctl reload nginx
+echo_progress_done "Nginx config applied"
+fi
```
Comment (flying-sausages) [inline scripts/install/komga.sh:52]: Might want to add an `else` statement for when nginx is not installed (just `echo_info` the port to the user)

**Review by flying-sausages — COMMENTED** [review]
Some minor comments.

File: scripts/install/komga.sh (hunk 4 above)
Comment (ComputerByte) [reply scripts/install/komga.sh:2]: Noted!

**Review by ComputerByte — COMMENTED** [review]
all fixed up, i think.

**Comment by flying-sausages** [comment]
There's a dependency on Java that needs to still be handled (there's #460 which handles that but I might need to split it off from airsonic). I'll take a look at that later on today

**Comment by flying-sausages** [comment]
Also, it needs a pass through `shfmt`, or you can use the vscode plugins we have listed in our CONTRIBUTORS.md

**Comment by ComputerByte** [comment]
oops, all done.

File: scripts/install/komga.sh [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+SERV
+echo_progress_done "Komga service installed"
+
+# This checks if nginx is installed, if it is, then it will install nginx config for komga
+if [[ -f /install/.nginx.lock ]]; then
+    echo_progress_start "Installing nginx config"
+    cat >/etc/nginx/apps/komga.conf <<-NGX
+location /komga {
+  include /etc/nginx/snippets/proxy.conf;
+  proxy_pass        http://127.0.0.1:8080/komga;
```
Comment (flying-sausages) [inline scripts/install/komga.sh:40]: `8080` will definitely conflict with something, we should make sure this port is changed

**Review by ComputerByte — COMMENTED** [review]
fixed

**Review by ComputerByte — COMMENTED** [review]
Nice

**Review by flying-sausages — APPROVED** [review]
Installs, works and all

**Pull request merged**